src/olympia/addons/models.py

```diff
@@ -541,6 +548,35 @@ def from_upload(cls, upload, platforms, source=None, is_listed=True,
 
         return addon
 
+    @classmethod
+    def resolve_webext_translations_and_save(cls, addon, data, upload):
+        """Resolve all possible translations from an add-on."""
+        fields = ('name', 'homepage', 'summary')
+        messages = extract_translations(upload)
+
+        # This is a bit stupid but I don't see any other way.
+        # We cannot attach actual translations while the add-on isn't saved
```

Comment:
Yes, I cannot set translation objects if the `addon` object is not yet saved (doesn'T have an id) - that's the cause for the error I described earlier (`1062, "Duplicate entry '1-en-US' for key 'id'"`)